### PR TITLE
Converted synthetic membership to natuaral for board's channel members

### LINKED ISSUE #3369: Bug: Cannot add explicit board members when a channel is linked to a board
## Steps to reproduce the behavior

1. Go to a board thats linked to a channel
2. Open Share modal
3. Add a team member to the board
4. See error - user is not added
5. Remove channel as a member of the board
6. Add the same team member from Step 3 - member can now be added

## Expected behavior

Users can be added to a board regardless if a channel is linked to the board or not.

## Screenshots (optional)

https://user-images.githubusercontent.com/93531870/179058094-dc834cfa-e106-480b-9c51-c4ad94e5bf48.mov

## Edition and Platform

 - Edition: Mattermost Boards (plugin)
 - Version: v7.2 Bug Bash
 - Browser and OS: Chrome on Mac

### PR DESCRIPTION
**Pull request by harshilsharma63**
Fixes #3369

### CODE REVIEW

**Review by mgdelacroix — CHANGES_REQUESTED** [review]
Code is looking good! could you please add an integration test for the use case?

**Comment by harshilsharma63** [comment]
@mgdelacroix I've added unit test for app later to cover this case.

File: server/app/boards_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+		th.Store.EXPECT().GetBoard(boardID).Return(&model.Board{
+			TeamID: "team_id_1",
+		}, nil)
+
+		th.Store.EXPECT().GetMemberForBoard(boardID, userID).Return(&model.BoardMember{
+			UserID:    userID,
+			BoardID:   boardID,
+			Synthetic: true,
+		}, nil)
```
Comment (mgdelacroix) [inline server/app/boards_test.go:89]: nit: you can return directly the board member you've created:

```suggestion
		th.Store.EXPECT().GetMemberForBoard(boardID, userID).Return(boardMember, nil)
```

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1: Minor, nonblocking nit

**Comment by harshilsharma63** [comment]
> LGTM! 👍 Minor, nonblocking nit

Actually, the two are not the same. One is synthetic and the other isn't.

**Pull request merged**